rust-runtime/aws-smithy-query/src/lib.rs

```diff
@@ -132,7 +132,11 @@ impl<'a> QueryListWriter<'a> {
     }
 
     pub fn finish(self) {
-        // Calling this drops self
+        // https://github.com/awslabs/smithy/commit/715b1d94ab14764ad43496b016b0c2e85bcf1d1f
+        // If the list was empty, just serialize the parameter name
+        if self.next_index == 1 {
```

Comment:
Not required since this is covered by protocol test, but it would be nice to have a Rust unit test for this.